### PR TITLE
EOS-17123: add draft component files

### PR DESCRIPTION
**Pull request by azheregelya**
Signed-off-by: Andrei Zheregelia <andrei.zheregelia@seagate.com>

## Problem Statement
<pre>
  <code>
 https://jts.seagate.com/browse/EOS-17123
 Draft component files for HA.
  </code>
</pre>
## Unit testing on RPM done
<pre>
  <code>
 No
  </code>
</pre>
## Problem Description
<pre>
  <code>
To validate whether cluster configuration conforms the reference we need this configuration to be formally defined.
Since it is planned to use component files in the future to generate cluster configuration, draft versions of such files can be used for validation purposes. It is enough to specify the name of resource, resource agent and its type. Other information can be added later once needed.
  </code>
</pre>
## Solution
<pre>
  <code>
These files are supposed to define cluster layout: resources to be
created, where they are located, what types do they have.

Currently this code is required only for cluster validation.
Later it will be used to generate cluster configuration.
  </code>
</pre>
## Unit Test Cases
<pre>
  <code>
Unit test is implemented to check whether these files can be parsed to ClusterLayout object.
  </code>
</pre>

### CODE REVIEW

File: conf/etc/v2/components/csm.json [hunk 1]
```diff
@@ -0,0 +1,161 @@
+{
+    "csm-agent": {
```
Comment (ajay-paratmandali) [inline conf/etc/v2/components/csm.json:2]: Is this file for test only ?
if in test we want to keep separate from main code

can we create separate dir
test/conf/    ?

Comment (azheregelya) [reply conf/etc/v2/components/csm.json:2]: I was considering these files as drafts for real component files. So, they are not only for tests.
Only limited amount of information is read for validation so far. The configuration provided in these files is severely incomplete (constrains are missing at least).

**Review by ajay-paratmandali — APPROVED** [review]

**Review by indrajitzagade — APPROVED** [review]

**Comment by priyankdal** [comment]
test this please

**Comment by gowthamchinna** [comment]
test this please

**Comment by gowthamchinna** [comment]
test this please

**Comment by ajaysrivas** [comment]
Merging this. Review comments will be taken care in https://jts.seagate.com/browse/EOS-20652.

**Comment by ajaysrivas** [comment]
Being merged because @azheregelya is on leave.

**Pull request merged**